src/test/org/codehaus/groovy/classgen/asm/sc/bugs/Groovy9892Bug.groovy

```diff
@@ -0,0 +1,21 @@
+package org.codehaus.groovy.classgen.asm.sc.bugs
+
+import groovy.transform.stc.StaticTypeCheckingTestCase
+import org.codehaus.groovy.classgen.asm.sc.StaticCompilationTestSupport
+
+class Groovy9892Bug extends StaticTypeCheckingTestCase implements StaticCompilationTestSupport {
+    void testUnaryOperationInGString() {
+        assertScript '''
+            class OuterClass {
+                int counter
+            
+                def call() {
+                    { ->
+                        "Hello${++counter}World"
+                    }.call()
+                }
+            }
+            assert new OuterClass().call() == 'Hello1World'
```

Comment:
Could you add a test case to cover `counter++` ?

Reply:
Test modified to verify both prefix and postfix increment